docs/src/explanations/dpi.md

```diff
@@ -0,0 +1,143 @@
+---
+layout: docs
+title:  "Calling Native Functions from Chisel (DPI)"
+section: "chisel3"
+---
+
+# Calling Native Functions from Chisel (DPI)
+
+## DPI Basics
+
+DPI API allows you to integrate native code into your Chisel hardware designs. This enables you to leverage existing libraries or implement functionality that is difficult to express directly in Chisel.
+
+Here's a simple example that demonstrates printing a message from a C++ function:
+```c++
+extern "C" void hello()
+{
+    std::cout << "hello from c++\\n";
+}
+```
+
+To call this function from Chisel, we need to define a corresponding DPI object.
+
+```scala
+object Hello extends DPIVoidFunctionImport {
+  override val functionName = "hello"
+  override val clocked = true
+  final def apply() = super.call()
+}
+
+Hello() // Print
+```
+
+Explanation:
+
+* `Hello` inherits from `DPIVoidFunctionImport` because the C++ function doesn't return a value (void).
+* `functionName` specifies the C-linkage name of the C++ function.
+* `clocked = true` indicates that its function call is invoked at clock's posedge.
+* We recommend defining the apply method for a more Scala-like syntax.
```

Comment:
Rather than saying "we recommend", just point out that you did this.

"We define an `apply` method for a function-like syntax. This allows us to call the DPI function with `Hello()`."